packages/synapse-react-client/src/components/SynapseChat/SynapseChatInteraction.tsx

```diff
@@ -19,14 +19,25 @@ import {
 import { Color } from '@mui/material/styles'
 import { TraceEvent } from '@sage-bionetworks/synapse-types'
 import { useEffect, useMemo, useRef, useState } from 'react'
+import { useNavigate } from 'react-router'
 import MarkdownSynapse from '../Markdown/MarkdownSynapse'
 
+// Hook to safely use navigate only when in a Router context
+function useSafeNavigate() {
+  try {
+    return useNavigate()
+  } catch {
+    return null
+  }
+}
+
```

Comment:
This breaks the rules of hooks. Could we instead explicitly pass information/a function via context?